Pinta/Dialogs/DocumentPropertiesDialog.cs

```diff
@@ -0,0 +1,202 @@
+            
+            initial_properties = new DocumentProperties(
+                author,
+                title,
+                subject,
+                keywords,
+                comments);
+            
+            authorEntry.Text = initial_properties.Author;
+            titleEntry.Text = initial_properties.Title;
+            subjectEntry.Text = initial_properties.Subject;
+            keywordsEntry.Text = initial_properties.Keywords;
+            commentsTextView.Buffer.Text = initial_properties.Comments;
+                        
+            authorEntry.Changed += OnAuthorChanged;
+            titleEntry.Changed += OnTitleChanged;
+            subjectEntry.Changed += OnSubjectChanged;
+            keywordsEntry.Changed += OnKeywordsChanged;
+            commentsTextView.Buffer.Changed += OnCommentsChanged; 
+            
+            AlternativeButtonOrder = new int[] { (int) Gtk.ResponseType.Ok, (int) Gtk.ResponseType.Cancel };
+            DefaultResponse = Gtk.ResponseType.Ok;
+        }
+        
+        public bool AreDocumentPropertiesUpdated {
+            get {
+                return initial_properties.Author != author
+                    || initial_properties.Title != title
+                    || initial_properties.Subject != subject
+                    || initial_properties.Keywords != keywords
+                    || initial_properties.Comments != comments;
+            }
+        }
+
+        public DocumentProperties InitialDocumentProperties {
+            get {
+                return initial_properties;
+            }
+        }
+        
+        public DocumentProperties UpdatedDocumentProperties {
+            get {
+                return new DocumentProperties (author, title, subject, keywords, comments);
+            }
+            set {
+                authorEntry.Text = author;
+                titleEntry.Text = title;
+                subjectEntry.Text = subject;
+                keywordsEntry.Text = keywords;
+                commentsTextView.Buffer.Text = comments;
+            }
+        }
+        
+        private void OnAuthorChanged (object sender, EventArgs e)
+        {
+            author = authorEntry.Text;
+        }
+
+        private void OnTitleChanged (object sender, EventArgs e)
+        {
+            title = titleEntry.Text;
+        }
+
+        private void OnSubjectChanged (object sender, EventArgs e)
+        {
+            subject = subjectEntry.Text;
+        }
+
+        private void OnKeywordsChanged (object sender, EventArgs e)
+        {
+            keywords = keywordsEntry.Text;
+        }
+
+        private void OnCommentsChanged (object sender, EventArgs e)
+        {
+            comments = commentsTextView.Buffer.Text;
+        }
+        
+        private void Build ()
+        {
+            DefaultWidth = 400;
+            DefaultHeight = 300;
+            BorderWidth = 6;
+
+            VBox tBox = new VBox ();// HACK handwriting GUI code sucks
```

Comment:
I'm not clear why is this needed, versus just using the VBox property of the dialog?